src/Knp/Snappy/Filesystem.php

```diff
@@ -0,0 +1,182 @@
+     */
+    public function __construct(string $temporaryDirectory = null)
+    {
+        $this->temporaryDirectory = $temporaryDirectory ?: sys_get_temp_dir();
+
+        register_shutdown_function([$this, 'removeTemporaryFiles']);
+    }
+
+    public function __destruct()
+    {
+        $this->removeTemporaryFiles();
+    }
+
+    /**
+     * Removes all temporary files.
+     */
+    public function removeTemporaryFiles()
+    {
+        foreach ($this->temporaryFiles as $file) {
+            if (!$this->exists($file)) {
+                continue;
+            }
+
+            $this->unlink($file);
+        }
+    }
+
+    /**
+     * Checks if $path exists.
+     *
+     * @param string $path
+     *
+     * @return bool
+     */
+    public function exists(string $path): bool
+    {
+        return file_exists($path);
+    }
+
+    /**
+     * Check if $path is a file.
+     *
+     * @param string $path
+     *
+     * @return bool
+     */
+    public function isFile(string $path): bool
+    {
+        return is_file($path);
+    }
+
+    /**
+     * Check if $path is a directory.
+     *
+     * @param string $path
+     *
+     * @return bool
+     */
+    public function isDir(string $path): bool
+    {
+        return is_dir($path);
+    }
+
+    /**
+     * Get the content of the file at $path.
+     *
+     * @param string $path
+     *
+     * @throws FileNotFound
+     *
+     * @return string
+     */
+    public function getFileContents(string $path): string
+    {
+        if (!$this->exists($path)) {
+            throw FileNotFound::notFound($path);
+        }
+
+        return file_get_contents($path);
+    }
+
+    /**
+     * Get the size of the file at $path.
+     *
+     * @param string $path
+     *
+     * @throws FileNotFound
+     *
+     * @return int
+     */
+    public function getFileSize(string $path): int
+    {
+        if (!$this->exists($path)) {
+            throw FileNotFound::notFound($path);
+        }
+
+        return filesize($path);
+    }
+
+    /**
+     * Deletes a file.
+     *
+     * @param string $filename
+     */
+    public function unlink(string $filename)
+    {
+        if (!$this->exists($filename)) {
+            return;
+        }
+
+        unlink($filename);
+
+        if ($this->exists($filename)) {
+            throw new \RuntimeException(sprintf('Could not delete already existing output file \'%s\'.', $filename));
+        }
+    }
+
+    /**
+     * Creates a directory.
+     *
+     * @param string $pathname
+     */
+    public function mkdir(string $pathname)
+    {
+        @mkdir($pathname, 0777, true);
+
+        if (!$this->isDir($pathname)) {
+            throw new \RuntimeException(sprintf('Unable to create directory "%s".', $pathname));
+        }
+    }
+
+    /**
+     * @param string $content
+     * @param string $extension
+     *
+     * @return string
+     */
+    public function createTemporaryFile($content = null, $extension = null): string
```

Comment:
Maybe we could use http://php.net/manual/fr/function.tmpfile.php instead ?

Reply:
`tmpfile` create a resource, we don't want that.